### PR TITLE
Noniid config

### PR DESCRIPTION
**Pull request by joyce-yuan**
Can now train with non iid test distribution. Also added fed weight and fed data repr algos into the algos section.

### CODE REVIEW

**Comment by tremblerz** [comment]
@joyce-yuan can you please fix the conflict

**Comment by joyce-yuan** [comment]
> @joyce-yuan can you please fix the conflict

yup just fixed

**Comment by tremblerz** [comment]
@joyce-yuan can you please move multiple system configurations in the sys_config.py

File: src/configs/algo_config.py [hunk 1]
```diff
@@ -131,4 +131,23 @@
     "exp_keys": ["similarity_metric", "selection_strategy", "consensus"]
 }
```
Comment (joyce-yuan) [inline src/configs/algo_config.py:133]: @nikitaparate for running a new algo, you just want to do something similar here where you port over the object from non iid config to algo_config and sys_config

**Review by tremblerz — APPROVED** [review]

**Pull request merged**